addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -110,15 +110,11 @@ def _check_move_configuration(self, move):
         if not re.match("^.{1,16}$", is_purchase and move.ref or move.name):
             error_message.append(_("%s number should be set and not more than 16 characters",
                 (is_purchase and "Bill Reference" or "Invoice")))
-        for line in goods_lines:
-            if line.display_type == 'product':
-                hsn_code = self._l10n_in_edi_extract_digits(line.l10n_in_hsn_code)
-                if not hsn_code:
-                    error_message.append(_("HSN code is not set in product line %s", line.name))
-                elif not re.match(r'^\d{4}$|^\d{6}$|^\d{8}$', hsn_code):
-                    error_message.append(_(
-                        "Invalid HSN Code (%s) in product line %s", hsn_code, line.name
-                    ))
+        product_lines = move.line_ids.filtered(lambda line: line.display_type == 'product' and not self._l10n_in_is_global_discount(line))
```

Comment:
why wouldn't we continue iterating over good_lines ? Do we want this behavior change ?

Reply:
Hello @casp-odoo ,
As per this task https://www.odoo.com/odoo/project/967/tasks/4032742?cids=2 from now we want to show hsn warning for all types of products.

Reply:
The task is marked as cancelled though ? 
In any case, such a functional change should be its own PR and is not related to this one.

Reply:
yes, PO cancelled that task because its spec is added to this task. and we are already working on that HSN alert part so they merged both tasks into one.

Reply:
Alright, I've had a look at both tasks, and although I understand what's going on it lacks a bit of clarity and written trace.
There seems to be no clear mention of the fusion of the tasks beside a comment on one of them. The main text of the tasks should feature this information. 

I still don't think this PR is an adequate environment to do this change ; but if it must, at the very least, the PR description should mention it clearly and it should be done in a separate commit with a clear description.